src/main/scala/com/gd/twitteranalytics/reports/TwitterReport.scala

```diff
@@ -9,8 +11,22 @@ trait TwitterReport {
   val today = java.time.LocalDate.now
   val dataPath = AppConfigReader.getAppConfigurables(0) + "/event_date=" + AppConfigReader.getReportConfigurables(1)
   val activeUserReportSavePath = AppConfigReader.getReportConfigurables(0) + "/ActiveUsers"
+  var errorMessage = ""
 
-  def getInputDataForReport(dataPath:String,spark:SparkSession):DataFrame = {
+  def validateReportPath(spark: SparkSession,log:Logger):Boolean = {
+    val fs = FileSystem.get(spark.sparkContext.hadoopConfiguration)
+    if (!fs.exists(new Path(dataPath))){
+      errorMessage = errorMessage.concat("Data for Report Creation is not available !")
+      false
+    }
+    else {
+      log.debug("======> Stage 1: Building Report for Twitter Performance...")
+      true
+    }
+  }
+
+  def getInputDataForActiveUserReport(dataPath:String,spark:SparkSession,log:Logger):DataFrame = {
+    log.debug("=======> Stage 2: Reading Data for Twitter Active User Report")
```

Comment:
Again... the order based logging should be in object that actually calls the functions, because it's the object who sets up the order.

Reply:
Fixed.